src/vs/workbench/contrib/debug/common/loadedScriptsPicker.ts

```diff
@@ -0,0 +1,109 @@
+ */
+export async function showLoadedScriptMenu(accessor: ServicesAccessor) {
+	const quickInputService = accessor.get(IQuickInputService);
+	const debugService = accessor.get(IDebugService);
+	const editorService = accessor.get(IEditorService);
+	const sessions = debugService.getModel().getSessions(false);
+	const modelService = accessor.get(IModelService);
+	const languageService = accessor.get(ILanguageService);
+	const labelService = accessor.get(ILabelService);
+
+	const localDisposableStore = new DisposableStore();
+	const quickPick = quickInputService.createQuickPick<IPickerLoadedScriptItem>();
+	localDisposableStore.add(quickPick);
+	quickPick.matchOnLabel = quickPick.matchOnDescription = quickPick.matchOnDetail = quickPick.sortByLabel = false;
+	quickPick.placeholder = nls.localize('moveFocusedView.selectView', "Search loaded scripts by name");
+	quickPick.items = await _getPicks(quickPick.value, sessions, editorService, modelService, languageService, labelService);
+
+	localDisposableStore.add(quickPick.onDidChangeValue(async () => {
+		quickPick.items = await _getPicks(quickPick.value, sessions, editorService, modelService, languageService, labelService);
+	}));
+	localDisposableStore.add(quickPick.onDidAccept(() => {
+		const selectedItem = quickPick.selectedItems[0];
+		selectedItem.accept();
+		quickPick.hide();
+		localDisposableStore.dispose();
+	}));
+	quickPick.show();
+}
+
+async function _getPicksFromSession(session: IDebugSession, filter: string, editorService: IEditorService, modelService: IModelService, languageService: ILanguageService, labelService: ILabelService): Promise<Array<IPickerLoadedScriptItem | IQuickPickSeparator>> {
+	const items: Array<IPickerLoadedScriptItem | IQuickPickSeparator> = [];
+	items.push({ type: 'separator', label: session.name });
+	const sources = await session.getLoadedSources();
+
+	sources.forEach((element: Source) => {
+		const pick = _createPick(element, filter, editorService, modelService, languageService, labelService);
+		if (pick) {
+			items.push(pick);
+		}
+
+	});
+	return items;
+}
+async function _getPicks(filter: string, sessions: IDebugSession[], editorService: IEditorService, modelService: IModelService, languageService: ILanguageService, labelService: ILabelService): Promise<Array<IPickerLoadedScriptItem | IQuickPickSeparator>> {
+	const loadedScriptPicks: Array<IPickerLoadedScriptItem | IQuickPickSeparator> = [];
+
+
+	const picks = await Promise.all(
+		sessions.map((session) => _getPicksFromSession(session, filter, editorService, modelService, languageService, labelService))
+	);
+
+	for (const row of picks) {
+		for (const elem of row) {
+			loadedScriptPicks.push(elem);
+		}
+	}
+	return loadedScriptPicks;
+}
+
+function _createPick(source: Source, filter: string, editorService: IEditorService, modelService: IModelService, languageService: ILanguageService, labelService: ILabelService): IPickerLoadedScriptItem | undefined {
```

Comment:
If you want, you can pass the ServicesAccessor around instead of passing a bunch of services to each function. Could be cleaner.

Reply:
I passed in the services because I was afraid that I would be fetching the service many times due to the number of scripts.